ui/raidboss/data/06-ew/ultimate/the_omega_protocol.ts

```diff
@@ -1381,6 +1381,369 @@ const triggerSet: TriggerSet<Data> = {
+          console.error(`Omega Safe Spots: missing m/f: ${JSON.stringify(data.combatantData)}`);
+          return;
+        }
+        if (f1.WeaponId === 4)
+          isF1In = true;
+        if (f2.WeaponId === 4)
+          isF2In = true;
+        if (m1.WeaponId === 4)
+          isM1In = true;
+        if (m2.WeaponId === 4)
+          isM2In = true;
+
+        if (isF1In)
+          distance1 = output.close!();
+        else if (isM1In)
+          distance1 = output.mid!();
+        else
+          distance1 = output.far!();
+
+        if (isF2In)
+          distance2 = output.close!();
+        else if (isM2In)
+          distance2 = output.mid!();
+        else
+          distance2 = output.far!();
+
+        // The combatants only spawn in these intercards:
+        // 92.93, 92.93 (NW)      107.07, 92.93 (NE)
+        // 92.93, 107.07 (SW)     107.07, 107.07 (SE)
+        // They will either spawn NW/SE first or NE/SW
+        // Boss cleave tells if it is actually east/west or north/south
+        if (matches.id === '7B9B') {
+          // East or West Safe
+          // Check for Sword/Shield to know if to go to Male or Female
+          const pos1 = (!isM1In && isF1In) ? f1.PosX : m1.PosX;
+          const pos2 = (!isM2In && isF2In) ? f2.PosY : m2.PosY;
+          dir1 = pos1 < 100 ? output.dirW!() : output.dirE!();
+          dir2 = pos2 < 100 ? output.dirN!() : output.dirS!();
+        } else {
+          // North or South Safe
+          const pos1 = (!isM1In && isF1In) ? f1.PosY : m1.PosY;
+          const pos2 = (!isM2In && isF2In) ? f2.PosX : m2.PosX;
+          dir1 = pos1 < 100 ? output.dirN!() : output.dirS!();
+          dir2 = pos2 < 100 ? output.dirW!() : output.dirE!();
+        }
+
+        // Secondary Spot for Staff + Sword
+        if (!isM1In && !isF1In) {
+          if (matches.id === '7B9B') {
+            // East/West Safe
+            if (f1.PosX < 100 && f1.PosY < 100) {
+              // NW
+              dir3 = output.dirWSW!();
+            } else if (f1.PosX < 100 && f1.PosY > 100) {
+              // SW
+              dir3 = output.dirWNW!();
+            } else if (f1.PosX > 100 && f1.PosY < 100) {
+              // NE
+              dir3 = output.dirESE!();
+            } else {
+              // SE
+              dir3 = output.dirENE!();
+            }
+          } else {
+            // North/South Safe
+            if (f1.PosX < 100 && f1.PosY < 100) {
+              // NW
+              dir3 = output.dirNNE!();
+            } else if (f1.PosX < 100 && f1.PosY > 100) {
+              // SW
+              dir3 = output.dirSSE!();
+            } else if (f1.PosX > 100 && f1.PosY < 100) {
+              // NE
+              dir3 = output.dirNNW!();
+            } else {
+              // SE
+              dir3 = output.dirSSW!();
+            }
+          }
+        }
+        if (!isM2In && !isF2In) {
+          if (matches.id === '7B9B') {
```

Comment:
Belatedly, is this logic incorrect? Should this boolean condition be flipped (i.e. `!== '7B9B'`) because it's for the second set?

I was going through and doing some edits/cleanup so that I could add a few extra output strings that some people requested and saw this.

Reply:
The output not matching the first set I think means the output is correct, but perhaps this was written poorly.

- The matches is based on the first cannon and the comment does not what is happening in the second cannon.


However, I think it is somehow possible to simplify the logic by placing it into a function since  7B9B M1+F1 === 7B9C M2 + F2.

Reply:
Ohhhh I see it's just the comment that's wrong.  I put it into a function in #5334.